tests/test_render.py

```diff
@@ -2275,3 +2281,265 @@ def set_bc(a):
         self.wait_for_text_to_equal(
             'body', 'Error loading dependencies', timeout=2
         )
+
+    # Clientside tests
+    def run_value_assertions(self, assertions):
+        for id in assertions:
+            WebDriverWait(self.driver, 10).until(
+                EC.text_to_be_present_in_element_value(
+                    (By.ID, id),
+                    assertions[id]
+                ),
+                message='Failed assertion: #{}.value != {}'.format(
+                    id, assertions[id]
+                )
+            )
+
+
+    def test_simple_clientside_serverside_callback(self):
+        app = dash.Dash(__name__, assets_folder='test_clientside')
+
+        app.layout = html.Div([
+            dcc.Input(id='input'),
+            html.Div(id='output-clientside'),
+            html.Div(id='output-serverside')
+        ])
+
+
+        @app.callback(
+            Output('output-serverside', 'children'),
+            [Input('input', 'value')])
+        def update_output(value):
+            return 'Server says "{}"'.format(value)
+
+
+        app.clientside_callback(
+            ClientsideFunction(
+                namespace='clientside',
+                function_name='display'
+            ),
+            Output('output-clientside', 'children'),
+            [Input('input', 'value')]
+        )
+
+        self.startServer(app)
+
+        input = WebDriverWait(self.driver, 10).until(
+            EC.presence_of_element_located((By.ID, "input"))
+        )
+        WebDriverWait(self.driver, 10).until(
+            EC.text_to_be_present_in_element(
+                (By.ID, 'output-serverside'),
+                'Server says "None"'
+            )
+        )
```

Comment:
Can we please add and use the methods we've already got in some of the other repos to reduce some of this repetition?

https://github.com/plotly/dash/blob/87091439ad0a5bb4abf336543e539aa4b0fd5e46/tests/IntegrationTests.py#L24-L33

Reply:
😸

Reply:
but these functions get marginally better every time i write a test....

Reply:
just kidding - 🔪 'ing away

Reply:
actually lol I spoke too soon. I had to modify the implementaiton under the hood a bit to handle `<input/>` `value`. So, it is truly a tiny bit better now. same API though!
Here's the commit - https://github.com/plotly/dash-renderer/pull/143/commits/374493df3403e7a5da55a73016f348d62d6dc2a3

Reply:
Ha, I didn't even see those ones that you modified - I guess sometimes we have these in the test file, sometimes in the base class... anyway yeah, nice tweak! (cc @byronz for whenever we manage to share these)